inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -410,8 +410,12 @@ def populate_submission_document(obj, eng):
             url=submission_pdf,
             original_url=submission_pdf,
         )
+
         obj.data = lb.record
 
+    if not obj.data.get('documents') and 'documents' in obj.data:
+        del obj.data['documents']
```

Comment:
I think that this check is very unclear, maybe if you really need to check explicitly if it's an empty list you can check:

```
if 'documents' in obj.data and len(obj.data['documents']) == 0:
```